c_src/quicer_connection.c

```diff
@@ -841,7 +841,14 @@ async_connect3(ErlNifEnv *env,
 
   // Monitor owner before start, so we don't need to race with callbacks
   // after start the connection
-  enif_monitor_process(NULL, c_ctx, &c_ctx->owner->Pid, &c_ctx->owner_mon);
+  //
+  if (!c_ctx->is_monitored
```

Comment:
Just noting: l. 1179 is slightly different: `(!c_ctx->is_monitored) && 0 == ...` versus `!c_ctx->is_monitored && 0 == ...`, in case C associativity rules make a difference here.